src/prune.ts

```diff
@@ -0,0 +1,186 @@
+import { join, sep as pathSeparator } from 'path'
+import Promise = require('any-promise')
+import { EventEmitter } from 'events'
+import { Emitter } from './interfaces'
+import { findProject } from './utils/find'
+import { readConfig, isFile, transformDtsFile, rimraf } from './utils/fs'
+import { CONFIG_FILE, TYPINGS_DIR, DTS_BROWSER_FILE, DTS_MAIN_FILE } from './utils/config'
+import { relativeTo, isDefinition } from './utils/path'
+
+const AMBIENT_TYPE_DIR = 'ambient'
+
+export interface PruneOptions {
+  cwd: string
+  production?: boolean
+  emitter?: Emitter
+}
+
+export function prune(options: PruneOptions): Promise<void> {
+  const { cwd, production } = options
+  const emitter = options.emitter || new EventEmitter()
+
+  return findProject(cwd)
+    .then((project: string) => {
+      return readMasterDependencies(project)
+        .then((masterDependencies: TypeDependencies) => {
+          const typingsPath = join(project, TYPINGS_DIR)
+
+          return Promise.all([
+            // Prune browser.d.ts
+            pruneExtraneous({
+              masterDependencies,
+              typingsPath,
+              includeDev: !production,
+              dtsPath: join(typingsPath, DTS_BROWSER_FILE),
+              emitter
+            }),
+            // Prune main.d.ts
+            pruneExtraneous({
+              masterDependencies,
+              typingsPath,
+              includeDev: !production,
+              dtsPath: join(typingsPath, DTS_MAIN_FILE),
+              emitter
+            })
+          ]).then(() => undefined)
+        })
+    })
+}
+
+interface TypeDependencies {
+  ambientDependencies?: { [type: string]: string }
+  ambientDevDependencies?: { [type: string]: string }
+  dependencies?: { [type: string]: string }
+  devDependencies?: { [type: string]: string }
+}
+
+function readMasterDependencies(projectDirectory: string): Promise<TypeDependencies> {
```

Comment:
You can also use https://github.com/typings/core/blob/master/src/utils/find.ts#L10 to find the config file.